Code/autopkglib/DmgMounter.py

```diff
@@ -162,19 +157,17 @@ def unmount(self, pathname):
 
         # Call hdiutil.
         try:
-            proc = subprocess.Popen(
+            proc = subprocess.run(
                 ("/usr/bin/hdiutil", "detach", self.mounts[pathname]),
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
+                capture_output=True,
                 text=True,
             )
-            (_, stderr) = proc.communicate()
         except OSError as err:
             raise ProcessorError(
                 f"hdiutil execution failed with error code {err.errno}: {err.strerror}"
             )
         if proc.returncode != 0:
-            raise ProcessorError(f"unmounting {pathname} failed: {stderr}")
+            raise ProcessorError(f"unmounting {pathname} failed: {proc.stderr}")
```

Comment:
Same feedback here as above - we should leverage check=True.

Reply:
e376acd